inst/stan/functions/log_expected_obs_from_r.stan

```diff
@@ -0,0 +1,21 @@
+  array[] vector log_expected_obs_from_r(
+    matrix lexp_obs_int, vector r, array[] int r_g, int r_t,
+    int r_seed, int gt_n, vector lrgt, int t, int g
+  ) {
+    array[g] vector[t] exp_lobs;
+    for (k in 1:g) {
+      vector[r_t] local_r = segment(r, r_g[k] + 1, r_t);
+      exp_lobs[k][1:r_seed] = lexp_obs_int[1:r_seed, k];
+      if (gt_n == 1) {
+        exp_lobs[k][(r_seed + 1):t] = exp_lobs[k][1] + cumulative_sum(local_r);
+      }else{
+        for (i in 1:r_t){
+          exp_lobs[k][r_seed + i] = local_r[i] +
+            log_sum_exp(
+              segment(exp_lobs[k], r_seed + i - gt_n, gt_n) + lrgt
+            );
```

Comment:
So this is modelling the log of the instantaneous reproduction number which when the generation time is 1 will be the log approximation to the growth rate. Are you modelling Rt on the natural scale directly in your your work?

Reply:
Ah okay, I was just confused here because everything is described as the growth rate and you used lower case `r` to denote this variable (and because the growth rate can of course be very different when we have an actual generation time distribution). Should we rename it to the reproduction number and use capital `R` instead then?

Nevertheless, we should decide whether the growth rate is also something we want to estimate/report alongside. If yes, it may make sense to model the growth rate and to compute R from it, because that is straightforward given a discrete generation time distribution, but not the other way round.

I have tried both: modeling Rt on the log scale and modeling Rt on the natural scale with a softplus link that avoids negative values.

Reply:
I think we may want to report the daily growth rate but perhaps we can do that as post-processing and save ourselves some hassle in stan? 

In terms of what we model it has always seemed natural to use a log link to me to make modifiers multiplicative. I know that the Imperial group have made a lot of use of a logit link (to provide upper and lower bounds). That seems fine but does perhaps introduce some quite unexpected behaviour for uses. 

How have you found the softplus to perform? I am in principle open to it (in the future multiple links seem like an obvious nice feature to have). In terms of what we name it I really don't mind. Perhaps it should be called R and then be made explicit somewhere what the link we are using is?

Reply:
Added an issue for looking at this (#176)